openapi/data_repository_service.swagger.yaml

```diff
@@ -194,57 +194,71 @@ definitions:
           sha512
   Bundle:
     type: object
-    required: ['id', 'object_ids', 'created', 'updated', 'version', 'checksums']
     properties:
       id:
         type: string
-        description: |-
+        description: >-
           An identifier, unique to this Data Bundle
       object_ids:
         type: array
         items:
           type: string
-        description: |-
+        description: >-
           The list of Data Objects that this Data Bundle contains.
       created:
         type: string
         format: date-time
-        description: |-
+        description: >-
           Timestamp of object creation in RFC3339.
       updated:
         type: string
         format: date-time
-        description: |-
-          Timestamp of update in RFC3339, identical to create timestamp in systems
-          that do not support updates.
+        description: >-
+          Timestamp of update in RFC3339, identical to create timestamp in
+          systems that do not support updates.
       version:
         type: string
-        description: |-
-          A string representing a version, some systems may use checksum, a RFC3339
-          timestamp, or incrementing version number. For systems that do not support
-          versioning please use your update timestamp as your version.
+        description: >-
+          A string representing a version, some systems may use checksum, a
+          RFC3339 timestamp, or incrementing version number. For systems that do not
+          support versioning please use your update timestamp as your version.
+      mime-type:
+        type: string
+        description: >-
+          A string providing the mime-type of the Data Bundle.
+          For example, "application/json".
       checksums:
         type: array
-        items:
-          $ref: '#/definitions/Checksum'
-        description: |-
+        description: >-
           At least one checksum must be provided.
           The Data Bundle checksum is computed over all the checksums of the
           Data Objects that bundle contains.
+        items:
+          $ref: '#/definitions/Checksum'
+      access_methods:
```

Comment:
Do we need `access_methods` for Bundles? Given that the payload of a bundle is the `contents` array, which is returned in the response to this method, what further access is needed?

Reply:
Yes, I think so. If the entire List of Data Objects in a Bundle existed in a FTP/S3/... directory, we would want to provide a `access url` to the whole directory. It would also help requesting access to the whole directory via a single `access_id`.

If such a directory does not exist, then the DRS provider simply provides an empty list of `access_methods`.

Reply:
I don't understand @susheel. You have stored the bundle in a file. But instead of producing that file contents as the response to this request, you want the client to make another request that will produce the Bundle response? So then you are not producing the contents array below?

Reply:
@ddietterich No, @dglazer's question is more about should Bundles have a`ccess_methods`. I believe the specification should not restrict this.
The `access_url` of a Bundle provides a way to access the Bundle `contents` byte array in a single call. Consider a folder example:
ftp://server.com/data/A
A
├── DO1
└── DO2
The Data Objects `DO1` & `DO2` will all have access_urls:
DO1.*.access_url: ftp://server.com/data/A/DO1
DO2.*.access_url: ftp://server.com/data/A/DO2

A client would hand these `access_url`s to ftp-client for download.

If I were to create a Bundle A with it's `contents` represented by `DO1` and `DO2`, then the question is should I be able to provide an access_url for the Bundle. My argument is that the specification should not restrict this use case.
A.access_url: ftp://server.com/data/A

A client would hand this `access_url` to ftp-client for download.

The service provider must be certain that directory structure provided by the `access_url` conforms to the structure described by the Bundle `contents`

Reply:
See https://github.com/ga4gh/data-repository-service-schemas/pull/244#issuecomment-480517411

Reply:
Removed `access_methods` to be discussed in a different PR.